.github/workflows/test.yml

```diff
@@ -22,23 +22,35 @@ jobs:
     needs: llvm-toolchain
     runs-on: ubuntu-latest
     outputs:
-      matrix: ${{ steps.set-matrix.outputs.matrix }}
+      build-matrix: ${{ steps.set-matrix-impl.outputs.build_matrix }}
+      test-matrix: ${{ steps.set-matrix-impl.outputs.test_matrix }}
     steps:
-      - id: set-matrix
+      - id: set-matrix-impl
+        shell: python3 -I {0}
         run: |
-          echo "::set-output name=matrix::{\"include\":[ \
-            {\"kernel\": \"LATEST\", \"runs_on\": [\"ubuntu-latest\", \"self-hosted\"], \"arch\": \"x86_64\", \"toolchain\": \"gcc\"}, \
-            {\"kernel\": \"LATEST\", \"runs_on\": [\"ubuntu-latest\", \"self-hosted\"], \"arch\": \"x86_64\", \"toolchain\": \"${{ needs.llvm-toolchain.outputs.llvm }}\"}, \
-            {\"kernel\": \"LATEST\", \"runs_on\": [\"z15\", \"self-hosted\"], \"arch\": \"s390x\", \"toolchain\": \"gcc\"} \
-          ]}"
-  VM_Test:
+          from json import dumps
+
+          matrix = [
+            {"kernel": "LATEST", "runs_on": ["ubuntu-latest", "self-hosted"], "arch": "x86_64", "toolchain": "gcc"},
+            {"kernel": "LATEST", "runs_on": ["ubuntu-latest", "self-hosted"], "arch": "x86_64", "toolchain": "${{ needs.llvm-toolchain.outputs.llvm }}"},
+            {"kernel": "LATEST", "runs_on": ["z15", "self-hosted"], "arch": "s390x", "toolchain": "gcc"},
+          ]
+          build_matrix = {"include": matrix}
+          print(f"::set-output name=build_matrix::{dumps(build_matrix)}")
+
+          tests = ["test_progs", "test_progs_noalu", "test_maps", "test_verifier"]
```

Comment:
as discussed, probably good to s/test_progs_noalu/test_progs_noalu32/ while it's not too late to keep it close to actual binary name

Reply:
I've renamed the test here https://github.com/kernel-patches/vmtest/pull/129.